returnn/tf/layers/basic.py

```diff
@@ -3414,35 +3406,48 @@ def get_out_data_from_opts(cls, name, axis, dims, pad_to_multiples=None, sources
     resolved_dims = None
     if dims and isinstance(dims[0], Dim):
       assert all(isinstance(d, Dim) for d in dims)
-      resolved_dims = dims
+      resolved_dims = tuple(dims)
       dims = [d.dimension or -1 for d in dims]
     if data.batch_shape[axis] is not None:
       resolved_shape_dims = cls._resolve_dims(
         old_dim=data.batch_shape[axis], new_dims=dims, pad_to_multiples=pad_to_multiples)
     else:
       resolved_shape_dims = tuple([(d if (d >= 0) else None) for d in dims])
 
+    axis_dim_tag = data.dim_tags[axis]
     rem_dim_indices = [i for i, d in enumerate(dims) if d < 0]
     assert len(rem_dim_indices) <= 1, "only one entry in dims %r can be -1" % (dims,)
-    rem_dim_idx = rem_dim_indices[0] if rem_dim_indices else (len(dims) - 1)
-    axis_dim_tag = data.dim_tags[axis]
-    if all(d == 1 for (i, d) in enumerate(dims) if i != rem_dim_idx):
-      rem_dim = axis_dim_tag  # we can overtake the existing dim tag
+    if len(rem_dim_indices) == 1:
+      import numpy
+      n = int(numpy.prod([dim for dim in dims if dim > 0]))
+      rem_dim_idx = rem_dim_indices[0]
+      if n == 1:
+        rem_dim = axis_dim_tag
+      else:  # need to create a new rem_dim
+        rem_dim = Dim(
+          kind=Dim.Types.Spatial,
```

Comment:
Not this?
```suggestion
          kind=axis_dim_tag.kind,
```

Reply:
Ah yeah, I missed that .. this is also the reason for the failing test.